mempool/mempool.go

```diff
@@ -577,7 +577,12 @@ func (cache *mapTxCache) Push(tx types.Tx) bool {
 
 	// Use the tx hash in the cache
 	txHash := sha256.Sum256(tx)
-	if _, exists := cache.map_[txHash]; exists {
+	if moved, exists := cache.map_[txHash]; exists {
+		//relocation the tx and push it to front
+		cache.list.Remove(moved)
+		delete(cache.map_, txHash)
+		cache.list.PushFront(txHash)
```

Comment:
Its possible to implement the LRU update much more efficiently. There is no need to update the map / remove the element from the list. Instead a linked list element can be moved by changing 6 pointers. Our linked list API should provide a way to do this. If it doesn't, we should adapt it so that it does.

Reply:
I found we use `golang/src/container/list` not `libs/clist` , and all of our write method in `mapTxCache`  use `sync.Mutex`,So I use `MoveToFront` method defined in `list.go`

https://github.com/golang/go/blob/859cf7fc0f4535ab3cdec15c81860f5fd2ae5b01/src/container/list/list.go#L165-L174